docs/data/joy/components/menu/MenuToolbarExample.tsx

```diff
@@ -0,0 +1,270 @@
+import * as React from 'react';
+import Menu from '@mui/joy/Menu';
+import MenuItem, { menuItemClasses } from '@mui/joy/MenuItem';
+import List from '@mui/joy/List';
+import ListItem from '@mui/joy/ListItem';
+import ListItemButton from '@mui/joy/ListItemButton';
+import ListDivider from '@mui/joy/ListDivider';
+import Typography, { typographyClasses } from '@mui/joy/Typography';
+import Dropdown from '@mui/joy/Dropdown';
+import MenuButton from '@mui/joy/MenuButton';
+import { Theme } from '@mui/joy';
+
+type MenuBarButtonProps = {
+  children: React.ReactNode;
+  open: boolean;
+  onOpen: (event?: React.KeyboardEvent | React.MouseEvent) => void;
+  onKeyDown: (event: React.KeyboardEvent) => void;
+  menu: JSX.Element;
+  onMouseEnter: React.MouseEventHandler;
+};
```

Comment:
We could simplify this and reuse `DropdownProps` from Joy UI:
```suggestion
type MenuBarButtonProps = Pick<DropdownProps, 'children' | 'open'> & {
  onOpen: DropdownProps['onOpenChange'];
  onKeyDown: React.KeyboardEventHandler;
  menu: JSX.Element;
  onMouseEnter: React.MouseEventHandler;
};
```

Reply:
`onOpen` signature is bit different than `onOpenChange` coming from `DropDownProps`. so used custom signature [here](https://github.com/mui/material-ui/pull/38308/files#diff-8770a67b556d81e12322c6f18c4088e3937b432ece0f1914229d9209de56e846R14) for `onOpen`. For remaining props, i've updated acc to your suggestion

Reply:
Actually arguments passed to `onOpen` is not used anywhere. so i defined it different. but it makes more sense to reuse type from `DropDownProps`. related commit https://github.com/mui/material-ui/pull/38308/commits/cafb973fb8e6f22d0af4cd345b930eeae5c87694

Reply:
Parameters passed could be used or not used when the callback is defined.